util/builtinobjects/builtinobjects.go

```diff
@@ -405,43 +420,92 @@ func (b *builtinObjects) getNewObjectID(spaceID string, oldID string) (id string
 	return "", err
 }
 
-func (b *builtinObjects) downloadZipToFile(url string) (path string, err error) {
+func (b *builtinObjects) downloadZipToFile(url string, progress process.Progress) (path string, err error) {
 	if err = uri.ValidateURI(url); err != nil {
 		return "", fmt.Errorf("provided URL is not valid: %w", err)
 	}
 	if !gallery.IsInWhitelist(url) {
 		return "", fmt.Errorf("URL '%s' is not in whitelist", url)
 	}
-	var resp *http.Response
-	// nolint: gosec
-	resp, err = http.Get(url)
+
+	var (
+		countReader *datacounter.ReaderCounter
+		size        int64
+	)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	go func() {
```

Comment:
maybe start goroutine after countReader is initialized?

Reply:
Intorduced mutex to exclude lock on reader init